modules/openapi-generator/src/main/resources/cpp-restbed-server/model-source.mustache

```diff
@@ -18,20 +21,52 @@ namespace {{this}} {
 
 {{classname}}::{{classname}}()
 {
-    {{#vars}}{{^isContainer}}{{#isPrimitiveType}}m_{{name}} = {{{defaultValue}}};
-    {{/isPrimitiveType}}{{^isPrimitiveType}}{{#isString}}m_{{name}} = {{{defaultValue}}};
-    {{/isString}}{{#isDateTime}}m_{{name}} = {{{defaultValue}}};
-    {{/isDateTime}}{{/isPrimitiveType}}{{/isContainer}}{{/vars}}
+    {{#vars}}
+	{{^isContainer}}
+	{{#isPrimitiveType}}
+	m_{{name}} = {{{defaultValue}}};
+	{{/isPrimitiveType}}
+	{{^isPrimitiveType}}
+	{{#isString}}
+	m_{{name}} = {{{defaultValue}}};
+	{{/isString}}
+	{{#isDate}}
+	m_{{name}} = {{{defaultValue}}};
+	{{/isDate}}
+	{{#isDateTime}}
+	m_{{name}} = {{{defaultValue}}};
+	{{/isDateTime}}
+	{{#isEnum}}
+	m_{{enumName}} = { {{#allowableValues}}{{#enumVars}}{{^-first}}, {{/-first}}{{{value}}}{{/enumVars}}{{/allowableValues}} };
+    {{/isEnum}}
+	{{/isPrimitiveType}}
+	{{/isContainer}}
+	{{/vars}}
```

Comment:
Some schemas might contain self, cross or circular references. Which could initialize infinite objects when parsing the JSON, as objects inside members would also be initialized upon initialisation. For that reason I've decided to initialize models as obejcts when actually required in `fromPropertyTree()`.  Also most members will receive their `{{{defaultValue}}}` if not supplied in the JSON.

Reply:
> Some schemas might contain self, cross or circular references.

You're right, I forgot about that. I wonder if there is a requirement that self/cross/circular property cannot be set as required. Then some kind of optional class would help, but I guess we don't have such constraint.

Then I'm fine with it, I'm thinking if it's viable to implement copy/move constructors to avoid this problem with reference-copying.

Reply:
> I wonder if there is a requirement that self/cross/circular property cannot be set as required. 

At least in the OpenAPI-Specification or in the Generator there is none. Then again the restbed generator currently doesn't validate if a passed value is writeable/readable/required.